src/locksManager.js

```diff
@@ -54,4 +54,20 @@ var LocksManager = GObject.registerClass({
         const variant = new GLib.Variant('a{sb}', {locked: false});
         this._proxy.SetSync(lock, variant);
     }
+
+    getTrapSequence(lock) {
+        return this._proxy.getDictValueSync(lock, 'trap_sequence');
```

Comment:
I would try to make this async if it's feasible (it may not be.)

Additionally, we call it three times in a row, so at least perform only one sync call and save the value.